crates/forge_domain/src/timeout_config.rs

```diff
@@ -0,0 +1,10 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Default, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TimeoutConfig {
+    pub read_timeout: Option<u64>,
+    pub pool_idle_timeout: Option<u64>,
+    pub pool_max_idle_per_host: Option<usize>,
+    pub max_redirects: Option<usize>,
+}
```

Comment:
Don't keep the fields optional. Implement `Default` and then in infra, overwrite the default values if its available in `env`.